src/App/DefaultSettingsHandler.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2017-2023 Smart code 203358507
+
+const React = require('react');
+const {
+    withCoreSuspender,
+    useProfile,
+} = require('stremio/common');
+const useProfileSettingsInputs = require('stremio/routes/Settings/useProfileSettingsInputs');
```

Comment:
I would suggest to use the core service transport to update settings instead of importing useProfileSettingsInputs and add `profile.settings` as a dependency to the hook
```js
const { useServices } = require('stremio/services');

const DefaultSettingsHandler = () => {
    const profile = useProfile();
    const { core } = useServices();
  
    React.useLayoutEffect(() => {
        const searchParams = new URLSearchParams(window.location.search);
        if (searchParams.has('streamingServerUrl')) {
           core.transport.dispatch({
                action: 'Ctx',
                args: {
                    action: 'UpdateSettings',
                    args: {
                        ...profile.settings,
                        streamingServerUrl: searchParams.get('streamingServerUrl')
                    }
                }
            });
        }
    }, [profile.settings]);
 
    return null;
};
```